src/libraries/System.Data.Common/src/System/Xml/XmlDataDocument.cs

```diff
@@ -21,13 +18,13 @@ namespace System.Xml
     [Obsolete("XmlDataDocument class will be removed in a future release.")]
     public class XmlDataDocument : XmlDocument
     {
-        private DataSet _dataSet;
+        private DataSet _dataSet = null!;
 
-        private DataSetMapper _mapper;
-        internal Hashtable _pointers;         // Hastable w/ all pointer objects used by this XmlDataDocument. Hashtable are guaranteed to work OK w/ one writer and mutiple readers, so as long as we guarantee
+        private DataSetMapper _mapper = null!;
```

Comment:
Same here - use `[MemberNotNull]` on the Init method

Reply:
https://github.com/dotnet/runtime/pull/41541#discussion_r479605477